### ISSUE TITLE
[REVIEW]: DetectorChecker: analyzing patterns of defects in detector screens

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2474]
**Submitting author:** @tomaslaz (<a href="http://orcid.org/0000-0002-8351-9857">Tomas Lazauskas</a>)
**Repository:** <a href="https://github.com/alan-turing-institute/DetectorChecker" target ="_blank">https://github.com/alan-turing-institute/DetectorChecker</a>
**Version:** v1.1.0
**Editor:** @fboehm
**Reviewers:** @janfreyberg, @craddm
**Archive:** <a href="https://doi.org/10.5281/zenodo.4312870" target="_blank">10.5281/zenodo.4312870</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/b6b67cd22d488e7bfa42b4074bc4eda8/status.svg)](https://joss.theoj.org/papers/b6b67cd22d488e7bfa42b4074bc4eda8)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/b6b67cd22d488e7bfa42b4074bc4eda8"><img src="https://joss.theoj.org/papers/b6b67cd22d488e7bfa42b4074bc4eda8/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/b6b67cd22d488e7bfa42b4074bc4eda8/status.svg)](https://joss.theoj.org/papers/b6b67cd22d488e7bfa42b4074bc4eda8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/alan-turing-institute/DetectorChecker" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@janfreyberg, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @fboehm know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/alan-turing-institute/DetectorChecker">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tomaslaz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

@craddm , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @fboehm know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/alan-turing-institute/DetectorChecker">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tomaslaz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @janfreyberg it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1088/0031-9155/42/1/001 is OK
- 10.1002/9781118658222 is OK
- 10.1002/bimj.201600178 is OK

MISSING DOIs

- https://doi.org/10.1038/146150e0 may be missing for title: Statistical method from the viewpoint of quality control

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02474/joss.02474/10.21105.joss.02474.pdf)

**Comment by fboehm** [comment]
Thank you, @janfreyberg ! Please feel free to mention me if you have questions or difficulties. Thanks again!!

**Comment by fboehm** [comment]
@tomaslaz - it looks like you're missing a doi according to the examination from whedon.

MISSING DOIs

- https://doi.org/10.1038/146150e0 may be missing for title: Statistical method from the viewpoint of quality control

Please be sure to verify that this doi is accurate, then add it to your manuscript.

**Comment by fboehm** [comment]
@tomaslaz - I'm struggling to find a second reviewer. Are there any scholars whom you can suggest as reviewers? Thanks again!

**Comment by fboehm** [comment]
@tomaslaz - please disregard the last message. I've found a second reviewer.

**Comment by whedon** [comment]
OK, @craddm is now a reviewer

**Comment by tomaslaz** [comment]
> @tomaslaz - it looks like you're missing a doi according to the examination from whedon.
> 
> MISSING DOIs
> 
> * https://doi.org/10.1038/146150e0 may be missing for title: Statistical method from the viewpoint of quality control
> 
> Please be sure to verify that this doi is accurate, then add it to your manuscript.

The suggested DOI is for a paper, however, we cite a book.

**Comment by fboehm** [comment]
Thanks for double-checking the doi, @tomaslaz. We'll proceed without a doi for that reference.

**Comment by fboehm** [comment]
@craddm and @janfreyberg - just wanted to check in to see how the review is going. Please let me know if there's something that I can do to help. Thanks again!!

**Comment by fboehm** [comment]
Thank you, @craddm for mentioning the issue. @tomaslaz - Do you have any questions on how to proceed with revisions based on the open issue from @craddm?

@janfreyberg Is there anything that I can do to help as you review the submission? Do you have any questions about the checklist? Thanks again!!

**Comment by janfreyberg** [comment]
Hi all - apologies for the long delay in my review, I should have updated you earlier. I've just updated the checklist and created an issue on the DetectorChecker repo!

**Comment by fboehm** [comment]
Excellent - thanks @janfreyberg ! @tomaslaz - please let me know if you have any questions about the opened issues. Thanks again!!

**Comment by fboehm** [comment]
@tomaslaz - I see that the issue from @janfreyberg remains open. Do you have any questions on how to resolve it? I think that it's one of the last hurdles before we can publish this.

**Comment by fboehm** [comment]
@craddm - It looks like three boxes remain unchecked in your checklist. If you haven't had a chance to suggest improvements for the documentation and functionality claims, please feel free to do so. Thanks again!

**Comment by tomaslaz** [comment]
thank you @fboehm  and @janfreyberg @craddm for the suggestions and recommendations! We are planning to work on the changes and resolve the raised issues soon.

**Comment by fboehm** [comment]
Thank you, @tomaslaz ! Please let us know when you have addressed the issues.

Thanks again!!

**Comment by fboehm** [comment]
@tomaslaz - I've added the "paused' label to give you time to address the issues. Do you anticipate that you can address the concerns within the next month, ie, by Nov 23? I don't want to keep the reviewers on hold for too long. Thanks again!

**Comment by tomaslaz** [comment]
@fboehm - apologies for taking our time addressing the concerns. @OscartGiles, any updates?

**Comment by OscartGiles** [comment]
Dear @fboehm and @tomaslaz - I'm planning to work through these this week so we should have a response soon. All the best.

**Comment by fboehm** [comment]
Thanks so much, @tomaslaz and @OscartGiles ! Please let me know if you have any questions, and please feel free to use this review thread to discuss and clarify the needed edits.

**Comment by fboehm** [comment]
Hi, @OscartGiles and @tomaslaz - I just wanted to check in to see how you are progressing through the issues from reviewers. Is there anything that I might help with right now?

**Comment by OscartGiles** [comment]
Hi @fboehm, 18 days went fast. I'm partway through the revisions now and we'll have them finished and submitted next week. All the best

**Comment by fboehm** [comment]
@OscartGiles - excellent! Thank you again! Please know that I'm happy to discuss any edits that you're about to make, especially if you're unsure about how to proceed.

**Comment by fboehm** [comment]
@OscartGiles @tomaslaz - Thanks for your continued work on this. How are you progressing with the revisions? Is there something that I can help with?

**Comment by OscartGiles** [comment]
Dear @fboehm , sorry about the delay. I think the changes are now complete. 

Thanks very much @fboehm and @craddm for your comments, they were really useful. Hope all is well Matt! (we used to be in the same department at UoL). 

I have created a single PR with all the changes (https://github.com/alan-turing-institute/DetectorChecker/pull/116) but the description links to individual PRs which have more details about the changes and which issues they address.

How do progress from here @fboehm? Would you like the review the PRs before they are merged to master? There are small changes to the paper (@tomaslaz - you might need to resubmit). I'm also going to make a few minor changes to the WebApp now to correct for any changes in the R package's API.

All the best,

Oscar

**Comment by fboehm** [comment]
Thanks, @OscartGiles ! I suggest merging in the pull requests. Once that is done, the reviewers can complete their reviews. Once they've checked all boxes in the above checklists, we can re-generate the pdf of the article and go forward with the submission.

**Comment by fboehm** [comment]
removed the "paused" label because revisions were made.

**Comment by janfreyberg** [comment]
That's a tremendous amount of work! The changes look great.

**Comment by fboehm** [comment]
@janfreyberg and @craddm - The authors have implemented many of the suggestions that arose in the review. Thanks for your work on this. When you're comfortable doing so, please complete the review checklist. If the authors need to do additional revisions before you are comfortable checking the boxes, please elaborate on how they can meet the expectations. Thanks again!

**Comment by janfreyberg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: :point_left:

**Comment by craddm** [comment]
Hi all,

Looks great - happy with the changes that have been made.

All good here thanks @OscartGiles, and hope same for you :)

Cheers,
Matt

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@OscartGiles - I made a pull request with a minor spacing fix. Once we incorporate this fix, we can proceed towards publication.

**Comment by OscartGiles** [comment]
@fboehm - Thank you! I've merged it now.

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@OscartGiles - I've opened a pull request with a small addition; namely, I think that there should be a formal citation to Shiny. I've added the bib entry and placed the pandoc citation where it's needed. Once it's merged, I'll regenerate the pdf.

**Comment by OscartGiles** [comment]
@fboehm - Excellent. Thanks for making the change. I've merged in.

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02474/joss.02474/10.21105.joss.02474.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@tomaslaz and @OscartGiles - Please make a tagged release of the software package and archive it on a service like Zenodo. Then report in this thread the software version number and the doi for the archive. Thanks!

**Comment by tomaslaz** [comment]
@fboehm thank you. The version number is v1.1.0 and the doi is 10.5281/zenodo.4312870 

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4312870.svg)](https://doi.org/10.5281/zenodo.4312870)

**Comment by fboehm** [comment]
@whedon set 10.5281/zenodo.4312870 as doi

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by fboehm** [comment]
@whedon set 10.5281/zenodo.4312870 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4312870" target="_blank">10.5281/zenodo.4312870</a> is the archive.

**Comment by fboehm** [comment]
@whedon set v1.1.0 as version

**Comment by whedon** [comment]
OK. v1.1.0 is the version.

**Comment by fboehm** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1971

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1971, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1088/0031-9155/42/1/001 is OK
- 10.1002/9781118658222 is OK
- 10.1002/bimj.201600178 is OK

MISSING DOIs

- 10.1038/146150e0 may be a valid DOI for title: Statistical method from the viewpoint of quality control

INVALID DOIs

- None
```

**Comment by fboehm** [comment]
The whedon output with missing DOIs is for a journal article and not for the book that the authors cite.

**Comment by Kevin-Mattheus-Moerman** [comment]
@tomaslaz I am processing the final steps to accept this work in JOSS. There are some minor points that need work. 

- [ ] Please amend the archive metadata such that the title and author list and order match that of the paper (@fboehm you can check this for future reference). 

- [ ] Could you please add city information to your affiliations and also please spell out UK as United Kingdom?

**Comment by tomaslaz** [comment]
> @tomaslaz I am processing the final steps to accept this work in JOSS. There are some minor points that need work.
> 
> * [x]  Please amend the archive metadata such that the title and author list and order match that of the paper (@fboehm you can check this for future reference).
> * [x]  Could you please add city information to your affiliations and also please spell out UK as United Kingdom?

This has been done @Kevin-Mattheus-Moerman.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1991

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1991, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1088/0031-9155/42/1/001 is OK
- 10.1002/9781118658222 is OK
- 10.1002/bimj.201600178 is OK

MISSING DOIs

- 10.1038/146150e0 may be a valid DOI for title: Statistical method from the viewpoint of quality control

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@tomaslaz it looks like you have not completed point 2 of my request :point_up: in relation to the affiliations. It may feel double/excessive to add town names and spell out the country but we'd like to be more homogeneous in our approach across submissions.

**Comment by tomaslaz** [comment]
@Kevin-Mattheus-Moerman apologies, my mistake, this should be fixed now.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1088/0031-9155/42/1/001 is OK
- 10.1002/9781118658222 is OK
- 10.1002/bimj.201600178 is OK

MISSING DOIs

- 10.1038/146150e0 may be a valid DOI for title: Statistical method from the viewpoint of quality control

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1999

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1999, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1341490112199356416) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2000
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02474](https://doi.org/10.21105/joss.02474)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
Congratulations @tomaslaz ! Thanks @fboehm for editing and thanks @janfreyberg and @craddm for reviewing this work.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02474/status.svg)](https://doi.org/10.21105/joss.02474)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02474">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02474/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02474/status.svg
   :target: https://doi.org/10.21105/joss.02474
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02474/status.svg)](https://doi.org/10.21105/joss.02474)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
